Compilation error with @Wither - Output stream or writer has already been opened

### Expected Behavior

Use Wither without Build failing

### Actual Behaviour

I'm was working only with @Builder without any problem, I;ve tried to work with @Wither to avoid copy records and got error compilation. 

> Task :compileJava FAILED
1 actionable task: 1 executed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
**> java.lang.RuntimeException: java.io.IOException: Output stream or writer has already been opened.**

The error occurr when implement the interface *Wither and try to build

### Steps To Reproduce

1. Create new micronaut project with Gradle (Kotlin)
2. Add      annotationProcessor("io.micronaut.sourcegen:micronaut-sourcegen-generator-java")
    implementation("io.micronaut.sourcegen:micronaut-sourcegen-annotations")
3. Build
4. Create a record and add @Wither
5. Build
6. Implement the interface *Wither
7. Build and Error

### Environment Information

Operating System: Windows 11
Micronaut Version: 4.5.0
JDK Version: 21
Gradle 8.8

### Example Application

_No response_

### Version

4.5.0

--- a/sourcegen-generator/src/main/java/io/micronaut/sourcegen/generator/visitors/BuilderAnnotationVisitor.java
+++ b/sourcegen-generator/src/main/java/io/micronaut/sourcegen/generator/visitors/BuilderAnnotationVisitor.java
@@ -38,6 +38,8 @@ import io.micronaut.sourcegen.model.StatementDef;
 import io.micronaut.sourcegen.model.TypeDef;
 import io.micronaut.sourcegen.model.VariableDef;
 
+import java.util.HashSet;
+import java.util.Set;
 import javax.lang.model.element.Modifier;
 import java.io.IOException;
 import java.util.ArrayList;
@@ -52,13 +54,23 @@ import java.util.List;
 @Internal
 public final class BuilderAnnotationVisitor implements TypeElementVisitor<Builder, Object> {
 
+    private final Set<String> processed = new HashSet<>();
+
     @Override
     public @NonNull VisitorKind getVisitorKind() {
         return VisitorKind.ISOLATING;
     }
 
     @Override
+    public void start(VisitorContext visitorContext) {
+        processed.clear();
+    }
+
+    @Override
     public void visitClass(ClassElement element, VisitorContext context) {
+        if (processed.contains(element.getName())) {
+            return;
+        }
         String simpleName = element.getSimpleName() + "Builder";
         String builderClassName = element.getPackageName() + "." + simpleName;
 
@@ -125,6 +137,7 @@ public final class BuilderAnnotationVisitor implements TypeElementVisitor<Builde
         }
 
         ClassDef builderDef = builder.build();
+        processed.add(element.getName());
         context.visitGeneratedSourceFile(
             builderDef.getPackageName(),
             builderDef.getSimpleName(),

--- a/sourcegen-generator/src/main/java/io/micronaut/sourcegen/generator/visitors/WitherAnnotationVisitor.java
+++ b/sourcegen-generator/src/main/java/io/micronaut/sourcegen/generator/visitors/WitherAnnotationVisitor.java
@@ -37,6 +37,8 @@ import io.micronaut.sourcegen.model.StatementDef;
 import io.micronaut.sourcegen.model.TypeDef;
 import io.micronaut.sourcegen.model.VariableDef;
 
+import java.util.HashSet;
+import java.util.Set;
 import javax.lang.model.element.Modifier;
 import java.io.IOException;
 import java.util.ArrayList;
@@ -54,13 +56,23 @@ import java.util.function.Consumer;
 @Internal
 public final class WitherAnnotationVisitor implements TypeElementVisitor<Wither, Object> {
 
+    private final Set<String> processed = new HashSet<>();
+
     @Override
     public @NonNull VisitorKind getVisitorKind() {
         return VisitorKind.ISOLATING;
     }
 
     @Override
+    public void start(VisitorContext visitorContext) {
+        processed.clear();
+    }
+
+    @Override
     public void visitClass(ClassElement recordElement, VisitorContext context) {
+        if (processed.contains(recordElement.getName())) {
+            return;
+        }
         if (!recordElement.isRecord()) {
             throw new ProcessingException(recordElement, "Only records can be annotated with @Wither");
         }
@@ -154,6 +166,7 @@ public final class WitherAnnotationVisitor implements TypeElementVisitor<Wither,
         }
 
         InterfaceDef witherDef = wither.build();
+        processed.add(recordElement.getName());
         context.visitGeneratedSourceFile(
             witherDef.getPackageName(),
             witherDef.getSimpleName(),

